.cirrus.yml

```diff
@@ -48,7 +48,13 @@ env:  # Global defaults
 
 # https://cirrus-ci.org/guide/tips-and-tricks/#sharing-configuration-between-tasks
 filter_template: &FILTER_TEMPLATE
-  skip: $CIRRUS_REPO_FULL_NAME == "bitcoin-core/gui" && $CIRRUS_PR == ""  # No need to run on the read-only mirror, unless it is a PR. https://cirrus-ci.org/guide/writing-tasks/#conditional-task-execution
+  # No need to run on the read-only mirror, unless it is a PR.
+  # https://cirrus-ci.org/guide/writing-tasks/#conditional-task-execution
+  skip: $CIRRUS_REPO_FULL_NAME == "bitcoin-core/gui" && $CIRRUS_PR == ""
+  # Allow forks to skip CI runs when a branch is pushed.
+  # Use only_if rather than skip, because otherwise all jobs are marked as
+  # skipped when a pull request is first opened.
+  only_if: $NO_BRANCH != "true" || $CIRRUS_PR != ""
```

Comment:
In commit "ci: forks can opt-out of CI branch push" (8fbf7449af130126890932986a4d9c0446c2f9c8)

What is $NO_BRANCH? Is it a custom variable like $NO_ARM added later? Could it be documented like NO_ARM is?

Reply:
It's indeed a custom variable like `NO_ARM`. I moved the explanation of `NO_BRANCH` up in the documentation. Leaving only the explanation of `skip` vs. `only_if` here, which I also reworded.